packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -945,21 +943,9 @@ abstract class ResidentRunner {
   }
 
   bool hasDirtyDependencies(FlutterDevice device) {
-    /// When using the build system, dependency analysis is handled by build
-    /// runner instead.
-    if (experimentalBuildEnabled) {
-      return false;
-    }
-    final DartDependencySetBuilder dartDependencySetBuilder =
-        DartDependencySetBuilder(mainPath, packagesFilePath);
-    final DependencyChecker dependencyChecker =
-        DependencyChecker(dartDependencySetBuilder, assetBundle);
-    if (device.package.packagesFile == null || !device.package.packagesFile.existsSync()) {
-      return true;
-    }
-    final DateTime lastBuildTime = device.package.packagesFile.statSync().modified;
-
-    return dependencyChecker.check(lastBuildTime);
+    // When using the build system, dependency analysis is handled by build
+    // runner instead.
+    return !experimentalBuildEnabled;
```

Comment:
What is the meaning of this method saying that there are always dirty dependencies when experimental build is not enabled? From what I can see this is used as `applicationNeedsRebuild` parameter. Why do we always have to rebuild if we are not using experimental build?

Reply:
Originally I disabled it when the experimental build_runner changes are enabled due to the analyzer based dependency script not supporting the build_cache/multiroot scheme. Now that we have a non-analyzer solution, I can replace both cases with the ProjectFileInvalidator.